components/dependencies.mdx

```diff
@@ -1,59 +1,105 @@
+Now Architect is here to provide the same value for APIs and microservices.
 
-## Utilizing dependencies
+## Why use dependencies?
 
-```yaml
+Citing dependencies as part of your component is not the most natural way for developers to integrate with
+software made by other teams, but its also the only way to achieve [on-demand environments](/environments/pull-request)
+for [distributed applications](/architecture/microservices).
+
+Every time a component is deployed with Architect, it will automatically deploy (if needed) and connect to
+cited dependencies. Perhaps even more impressive than that is that it will also deploy the dependencies OF your
+component's dependencies and ensure they're integrated as well.
+
+Without dependency management integrated into deployments, each and every application would have to maintain a
+complex web of CI pipelines or centralized infrastructure-as-code (IaC) templates to create new environments which
+can be an educational and logistical nightmare. With Architect, all you ever need to know is your immediate dependencies
+and the rest will be automated for you.
+
+## Registering dependencies
+
+Before we can begin using dependencies, we need to create a component that we want to act as a dependency for another.
+Let's start by creating a new component that we'll call the `backend` of our project. The command below will help you
+create a new component from one of Architect's starter projects.
+
+```sh
+$ architect init backend
+```
+
+Now that we have the project created, let's go ahead and register it with Architect. This will simply build the services
+into docker images and publish everything to Architect's cloud registry.
+
+```sh
+$ architect register ./backend --tag latest --account my-account
+```
+
+## Using dependencies
+
+Now that we have a component in the registry, we can use it as a dependency for another one. Let's create a second project
+and call it `frontend`.
+
+```sh
+$ architect init frontend
+```
+
+Next, let's open up `./frontend/architect.yml` in our favorite IDE so we can add the dependency. First, let's cite the dependency
+so that the backend will be deployed alongside the frontend. Add the following to the component configuration:
+
+```yaml architect.yml
 dependencies:
-  authentication: latest
+  backend: latest
+```
 
+Next, we'll need to ensure the frontend service has access to the backend API. Find the `environment` variables of the `app` service
+and add the following:
+
+```yaml architect.yml
 services:
-  my-api:
-    interfaces:
-      http: 8080
+  app:
+    build:
+      context: ./
     environment:
-      AUTH_INTERNAL: ${{ dependencies['authentication'].services.auth.interfaces.main.url }}
-      AUTH_EXTERNAL: ${{
-        dependencies['authentication'].services.auth.interfaces.main.ingress.url
-        }}
+      BACKEND_ADDR: ${{ dependencies.backend.services.api.interfaces.main.url }}
```

Comment:
I would explain where this address comes from and possibly link to components/service-discovery

Reply:
I see you bring in service discovery below. It would still be helpful to explain here that the address comes from the backend app's service declaration in its own architect.yml file.